### DISCUSSION TITLE
Will this mod support Klipper 12 in the future?

### DISCUSSION THREAD

**Discussion by Bram-Brooks** [xblax/flashforge_ad5m_klipper_mod#159] [Q&A]
Fantastic project! Wondering if Klipper 12 will ever be a reality on the AD5M? Thanks in advance.

**Comment by consp** [comment]
> Wondering if Klipper 12 will ever be a reality on the AD5M?

If Flashforge releases the modified makefile for the nations stm clone-chip that will be a lot easier (they haven't).

or if @KaruroChori succeeds in building it without it.

edit: Converted to discussion as it's not really an issue.

**Comment by Bram-Brooks** [comment]
It seems highly unlikely that FF will ever release that. Fingers crossed @KaruroChori manages to figure it out. Thanks for all the effort guys, much appreciated.

**Comment by dylansison** [comment]
@KaruroChori Per [this printer.cfg’s first comment](https://github.com/Klipper3d/klipper/blob/master/config/printer-voxelab-aquila-2021.cfg), it seems like there is an option in Klipper’s makefile already to compile for N32G45x mcus

I dunno how to go about testing this right now as I’m not sure how to go about flashing from onboard the AD5M, but I wonder if you have tried this already and if so, what happened?

**Reply by KaruroChori** [reply]
Attempts from multiple people are documented @ https://github.com/xblax/flashforge_ad5m_klipper_mod/discussions/155

**Comment by KaruroChori** [comment]
Now in https://github.com/xblax/flashforge_ad5m_klipper_mod/releases/tag/v00.06-beta
